### ISSUE TITLE
[HOLD for payment 2023-07-18][$1000] Dev - Console warning on pressing any mentioned user from conversation

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#21506]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Login to ND
2. Go to any conversation
3. Type a text, mention any user in that text and send it (open console)
4. Click on that particular mentioned in the chat
5. Notice on clicking mentioned it throws warning

## Expected Result:
No console warning should appear

## Actual Result:
Console warning appears

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] MacOS / Chrome / Safari
- [x] MacOS / Desktop

**Version Number:** v1.3.29-0
**Reproducible in staging?:** n/a
**Reproducible in production?:** n/a
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation
![Screenshot 2023-06-19 at 2 53 17 PM](https://github.com/Expensify/App/assets/93399543/00d15f73-1be9-4e61-bd40-124e670fc1ff)

**Expensify/Expensify Issue URL:**
**Issue reported by:** @dhairyasenjaliya
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1687166767791559

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~010aec32706776696c</li>
        <li>Upwork Job ID: 1673995397535133696</li>
        <li>Last Price Increase: 2023-06-28</li>
    </ul>
</details>

**Comment by kbecciv** [comment]
### Proposal
### Please re-state the problem that we are trying to solve in this issue.
- Dev - Console warning on pressing any mentioned user from conversation
### What is the root cause of that problem?
- The root cause is we recently started using `<TextLink>` component for `MentionUserRenderer.js` instead of just `<Text>` so here the problem is we are not passing `accessibilityLabel` which required props that is the cause of the error as this should be mandatory as focusable is always true that required accessibilityLabel currently we are not passing or not even passing `accessible` to false that makes to throw warning on console
### What changes do you think we should make in order to solve the problem?
* We need to add `accessibilityLabel` in order to remove the error log this is mandatory because by default focusable is true so we do need to provide accessibilityLabel
* We need to define new accessibilityHints for the mentioned user which will be passed to `accessibilityLabel` message in both languages
* If we don’t need accessibilityLabel inside `MentionUserRenderer.js` then we can just pass accessible to false this will also remove the error
### Optional changes
- Disregards this error I have logged another bug report that on opening detail page it throws another warning of `Warning: Failed prop type: Invalid prop `style` of type `object` supplied to `PressableWithoutFocus`, expected an array.`
- If we need to handle it here I have made a Proposal here https://expensify.slack.com/archives/C049HHMV9SM/p1687335105698689?thread_ts=1687326565.653439&cid=C049HHMV9SM

**Comment by StevenKKC** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Console warning on pressing any mentioned user from conversation.

### What is the root cause of that problem?
User details is displayed in `DetailsPage` component, and `DetailsPage` contains `PressableWithoutFocus`.
https://github.com/Expensify/App/blob/820fa93ad1e799c12d8b639b98018b5350ea17af/src/pages/DetailsPage.js#L150-L152

And `PressableWithoutFocus` has `style` props with `array` type.
https://github.com/Expensify/App/blob/820fa93ad1e799c12d8b639b98018b5350ea17af/src/components/Pressable/PressableWithoutFocus.js#L19-L19

But `style` prop that is passed to `PressableWithoutFocus` is not `array` type, so the console warning is displayed.

### What changes do you think we should make in order to solve the problem?
We can change `style` prop to `array` type.
```
   <PressableWithoutFocus
      style={[styles.noOutline]}
      onPress={show}
   >
```
### What alternative solutions did you explore? (Optional)
None.

**Comment by bfitzexpensify** [comment]
Hey @cristipaval - I can't test on dev, can you confirm this is reproducible? Thanks!

**Comment by cristipaval** [comment]
I can't reproduce with the latest dev.

**Comment by bfitzexpensify** [comment]
@dhairyasenjaliya can you still reproduce this?

**Comment by bfitzexpensify** [comment]
Thanks @dhairyasenjaliya - @cristipaval, mind trying again?

**Comment by cristipaval** [comment]
alright, I could reproduce. Updated the reproducible steps to make clearer.

**Comment by hugolgst** [comment]
Contributor details
Your Expensify account email: hugo.lageneste@pm.me
Upwork Profile Link: https://www.upwork.com/freelancers/~01e31236d7967e4bc5

**Comment by mollfpr** [comment]
Will reviewing the proposal in the morning 🙏

**Comment by mollfpr** [comment]
Can anyone share the video to reproduce this issue? I can't reproduce it; maybe I missed something(?).

**Comment by cristipaval** [comment]
@mollfpr do you have some time to review the proposal or you want to wait a bit longer for other proposals?

**Comment by mollfpr** [comment]
Thanks for the ping @cristipaval 

I'm reviewing the proposal now.

**Comment by mollfpr** [comment]
@cristipaval I can't reproduce the accessibility error, but we still can fix the `style` error with @dhairyasenjaliya [proposal](https://github.com/Expensify/App/issues/21506#issuecomment-1605688932).

Should we proceed?

**Comment by cristipaval** [comment]
@bfitzexpensify I don't know why this wasn't automatically updated. The PR was deployed into production 2 days ago. We'll have to do the payments in 5 days.

**Comment by mollfpr** [comment]
> [@mollfpr] The PR that introduced the bug has been identified. Link to the PR:
 [@mollfpr] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:

I'm unable to track which PR that causing the console error; it's between the migration on `PressableWithoutFocus` https://github.com/Expensify/App/pull/20202 or the changes on the `TouchableWithoutFocus` to `PressableWithoutFocus` in https://github.com/Expensify/App/pull/8483

> [@mollfpr] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:

The `console.error` can be caught on the PR review. This issue might arise when migrating the generic component and the test is not happening in all the pages, but we have a checklist to check that.

>If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)

> [@mollfpr] Determine if we should create a regression test for this bug.
 [@mollfpr] If we decide to create a [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) for the bug, please propose the regression test steps to ensure the same bug will not reach production again.

The error is only shown in DEV, so no regression test is needed. The reviewer checklist already covers this.

**Comment by mollfpr** [comment]
Melvin did not calculate the date...

@dhairyasenjaliya Got assigned on **Jul 5**

![Screenshot 2023-07-18 at 14 07 36](https://github.com/Expensify/App/assets/25520267/d8c21205-f7ba-4318-89e4-53f4d833e03e)

And the PR merged on **Jul 6**
![Screenshot 2023-07-18 at 14 07 47](https://github.com/Expensify/App/assets/25520267/edc4cfd1-f0c6-4fb2-b378-6fc708bc0231)

So merged PR within 1.042 days 🚀

**Comment by bfitzexpensify** [comment]
Cool, all payments issued here, BZ checklist complete, so we're all done - thanks for the solid work!
